### PR TITLE
feat: Add support for WinUI 3.0 namespaces

### LINKED ISSUE #3109: WinUI 3.0 namespaces alignment
This issue tracks the updates needed to align Uno with WinUI 3.0 namespaces.

This repository will entirely be converted to WinUI 3.0 namespaces, and a new Uno.WinUI package will be created as a result. To preserve the backward compatibility, the existing Uno.UI package will still be generated through a CI phase and tooling that will convert back to the UWP/WinUI 2.0 namespaces where appropriate.

Additional details will be included in this issue as the implementation progresses.

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes #3109

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

This PR introduces the generation of nuget packages compatible WinUI 3.0. The Uno.UI packages are generated as part of the main build, and additional packages are built to be compatible with the new namespaces.

Note that this update only adds Uno definitions for the 19041 API, and does not bump the Samples app version, as the 19041 update is not out of preview as of this commit.

## Migrating an app to WinUI 3.0

- Nuget updates
    - `Uno.UI` becomes `Uno.WinUI`
    - `Uno.UI.RemoteControl` becomes `Uno.WinUI.RemoteControl`
    - `Uno.UI.Lottie` becomes `Uno.WinUI.Lottie`
    - `Uno.UI.DualScreen` becomes `Uno.WinUI.DualScreen`
- String replacements:
    - `Windows.UI.Xaml` becomes `Microsoft.UI.Xaml`
    - `Windows.UI.Composition` becomes `Microsoft.UI.Composition`
- Loggers updates
```
{ "Windows", LogLevel.Warning },
{ "Microsoft", LogLevel.Warning },
```

## API Changes as of Alpha builds

- `Thickness` is missing a constructor
- `Duration` is missing a constructor
- `GridLength` is missing a constructor
- `MapControl` is missing
- `WebView` is now `WebView2`

## Creating an application from the templates

A blank application can be created through:

```bash
dotnet new unoapp-winui
```

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWPSyncGenerator/Generator.cs [hunk 1]
```diff
@@ -793,6 +823,11 @@ protected void BuildMethods(INamedTypeSymbol type, IndentedStringBuilder b, Plat
 		{
 			foreach (var method in type.GetMembers().OfType<IMethodSymbol>())
 			{
+
+				if (method.ToString().Contains("GetStringForUri"))
+				{
```
[the remark this answers is not in the document]
Comment (davidjohnoliver) [reply src/Uno.UWPSyncGenerator/Generator.cs:828]: +1

File: src/Uno.WinUIRevert/Program.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			// ReplaceInFile(Path.Combine(basePath, @"src\Uno.UI\Microsoft\UI\Xaml\Controls\TwoPaneView\TwoPaneView.xaml"), "using:Windows.UI.Xaml.Controls", "using:Windows.UI.Xaml.Controls");
+
+			// Adjust Colors
+			ReplaceInFile(Path.Combine(basePath, @"src\Uno.UI\UI\Colors.cs"), "Windows.UI", "Microsoft.UI");
+			ReplaceInFile(Path.Combine(basePath, @"src\Uno.UI\UI\ColorHelper.cs"), "Windows.UI", "Microsoft.UI");
+			ReplaceInFile(Path.Combine(basePath, @"src\SourceGenerators\Uno.UI.SourceGenerators\XamlGenerator\XamlConstants.cs"), "Windows.UI", "Microsoft.UI");
+			ReplaceInFile(Path.Combine(basePath, @"src\Uno.UI\UI\Xaml\Markup\Reader\XamlConstants.cs"), "Windows.UI", "Microsoft.UI");
+
+			// Custom animation
+			// ReplaceInFile(Path.Combine(basePath, @"src\Uno.UI\UI\Xaml\Media\Animation\Animators\RenderingLoopAnimator.wasm.cs"), "Microsoft", "Windows");
```
Comment (davidjohnoliver) [inline src/Uno.WinUIRevert/Program.cs:72]: Was this commented code left in on purpose?

Comment (jeromelaban) [reply src/Uno.WinUIRevert/Program.cs:72]: It is. I know I'm going to need it at some point, it was for the reverse operation.

**Pull request merged**